### PR TITLE
Add flowMax type generator and types

### PR DESCRIPTION
**Pull request by peterstuart**
This adds a script can generate types for `flowMax` up to arbitrary arities (I have it set to 100 parameters for now.). Since lodash's `flow` only is typed up to 7 parameters, we will either need to override its type somehow, or publish our own version of `flow` that uses the same type as `flowMax`.

### CODE REVIEW

File: types/ad-hok/flowMax.d.ts [hunk 1]
```diff
@@ -0,0 +1,10329 @@
+// Generated with "yarn generateFlowMaxType"
```
Comment (peterstuart) [inline types/ad-hok/flowMax.d.ts]: I knew this would be big, but I was not expecting 10,000+ lines…

File: types/ad-hok/index.d.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/// <reference path="./flowMax.d.ts" />
```
Comment (helixbass) [inline types/ad-hok/index.d.ts]: Can you not use a relative `import` statement to do this (eg `import {FlowMaxType} from './flowMax.d.ts'`)?

File: scripts/generateFlowMaxType.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const typeParameter = n => `T${n}`
+const parameterName = n => `f${n}`
+
+const generate = n => {
+  if (n === 0) {
```
Comment (helixbass) [inline scripts/generateFlowMaxType.js]: Should include a signature for zero arguments ie `flowMax()`?

File: scripts/generateFlowMaxType.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+const typeParameter = n => `T${n}`
+const parameterName = n => `f${n}`
```
Comment (helixbass) [inline scripts/generateFlowMaxType.js]: Unused?

File: scripts/generateFlowMaxType.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+const arities = generate(100).join('\n')
+
+const interfaceString = `// Generated with "yarn generateFlowMaxType"
+
+interface FlowMaxType {
+${arities}
+}
+`
+
+console.log(interfaceString)
```
Comment (helixbass) [inline scripts/generateFlowMaxType.js]: What's the Node-y way to write to stdout?

File: types/ad-hok/index.d.ts (hunk 2 above)
Comment (peterstuart) [reply types/ad-hok/index.d.ts]: That doesn't seem to work when importing our own type definitions. 🤷‍♂

File: scripts/generateFlowMaxType.js (hunk 5 above)
Comment (peterstuart) [reply scripts/generateFlowMaxType.js]: `process.stdout.write`

**Pull request merged**